integration-tests/src/test/java/oracle/weblogic/kubernetes/ItRetryImprovementMisc.java

```diff
@@ -0,0 +1,471 @@
+    assertDoesNotThrow(() -> deletePod(podName, domainNamespace),
+        String.format("delete pod %s in namespace %s failed", podName, domainNamespace));
+
+    // verify both domain and cluster: Available should go false, Complete should also go false
+    checkDomainStatusClusterConditionTypeHasExpectedStatus(withQuickRetryPolicy, domainUid, domainNamespace,
+        clusterName, DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE, "False", DOMAIN_VERSION);
+    checkDomainStatusClusterConditionTypeHasExpectedStatus(withQuickRetryPolicy, domainUid, domainNamespace,
+        clusterName, DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "False", DOMAIN_VERSION);
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace,
+        DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "False");
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace,
+        DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE, "False");
+
+    // wait the pod is restarted and back to ready
+    checkPodReady(podName, domainUid, domainNamespace);
+
+    // verify both domain and cluster: Available should go true, Complete should also go true
+    checkDomainStatusClusterConditionTypeHasExpectedStatus(withQuickRetryPolicy, domainUid, domainNamespace,
+        clusterName, DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE, "True", DOMAIN_VERSION);
+    checkDomainStatusClusterConditionTypeHasExpectedStatus(withQuickRetryPolicy, domainUid, domainNamespace,
+        clusterName, DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "True", DOMAIN_VERSION);
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace,
+        DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "True");
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace,
+        DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE, "True");
+  }
+
+  /**
+   * Set cluster replicas=0. Cluster Available should go to false and stay false. Domain available should stay true
+   * throughout. Domain and Cluster Complete should go false while the cluster is shutting down,
+   * and go True once the shutdown completes
+   */
+  @Order(5)
+  @Test
+  @DisplayName("Set cluster replicas to 0 and verify the cluster and domain status")
+  void testClusterReplicasToZero() {
+    logger.info("patch the cluster resource with replicas to 0");
+    String patchStr = "[{\"op\": \"replace\", \"path\": \"/spec/replicas\", \"value\": 0}]";
+
+    logger.info("Updating cluster configuration using patch string: {0}", patchStr);
+    V1Patch patch = new V1Patch(patchStr);
+    assertTrue(patchClusterCustomResource(clusterName, domainNamespace,
+        patch, V1Patch.PATCH_FORMAT_JSON_PATCH), "Failed to patch cluster");
+
+    // check the domain Complete goes to false
+    checkDomainStatusConditionTypeHasExpectedStatus(shortRetryPolicy, domainUid, domainNamespace,
+        DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "False", DOMAIN_VERSION);
+
+    // Enable this check when OWLS-104622 is fixed
```

Comment:
No reference to OWLS in public code. You can update the JIRA how to verify the usecase after changing the code e.g add the following libe to tehi class to verify teh resolution 

checkDomainStatusClusterConditionTypeHasExpectedStatus(shortRetryPolicy, domainUid, domainNamespace,
   clusterName, DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "False", DOMAIN_VERSION)